apps/files/src/mixins.js

```diff
@@ -100,6 +100,16 @@ export default {
         })
       })
     },
+    compareIds (x, y) {
+      if (!isNaN(x)) { // OC10 autoincrement id
+        return parseInt(x) === parseInt(y)
```

Comment:
`parseInt(x, 10)`. without this leading zeroes would cause Javascript to parse numbers as octal...

Reply:
Ah, I was was aware of the radix problem but was blatantly relying on [ES5](https://www.ecma-international.org/ecma-262/5.1/#sec-15.1.2.2) definition:

> If radix is undefined or 0, it is assumed to be 10 except when the number begins with the character pairs 0x or 0X, in which case a radix of 16 is assumed. If radix is 16, the number may also optionally begin with the character pairs 0x or 0X.

the definition applies also on `Number` [ES6](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseInt#Description):

>This method has the same functionality as the global parseInt() function:
>
>Number.parseInt === parseInt; // true
>
>and is part of ECMAScript 2015 (its purpose is modularization of globals). Please see parseInt() for more detail and examples.

In any case, a check for hex would make it less error prone, but octal should not be a concern, unless someone's using a version older than ES5... .

In any case, to make things clear I'd add the radix 💃